tools/base/src/Base.ts

```diff
@@ -84,18 +84,30 @@ export function SpectrumMixin<T extends Constructor<ReactiveElement>>(
             if (!activeElement) {
                 return false;
             }
+
             // Browsers without support for the `:focus-visible`
             // selector will throw on the following test (Safari, older things).
             // Some won't throw, but will be focusing item rather than the menu and
             // will rely on the polyfill to know whether focus is "visible" or not.
             try {
-                return (
+                return !!(
                     activeElement.matches(':focus-visible') ||
-                    activeElement.matches('.focus-visible')
+                    activeElement.matches('.focus-visible') ||
+                    activeElement.shadowRoot?.activeElement?.matches(
+                        ':focus-visible'
+                    ) ||
+                    activeElement.shadowRoot?.activeElement?.matches(
+                        '.focus-visible'
+                    )
```

Comment:
This doesn't _feel_ like the right fix. Namely in that it changes the meaning of the method from it's current name, but also because this move to encapsulation breaking which isn't something we should do lightly.

I'd want to look into why the visible focus is resolved on `Shift + Tab` entrance of the element and not on the `Tab` entrance before committing to something like this. Hopefully there's some less invasive insight to be gained from that...

Reply:
When tabbing forward, the `activeElement` will be the `sp-color-slider` or `sp-color-wheel`, but that element does not match `:focus-visible` or `.focus-visible`, because it is not the actual element within its shadowRoot that has focus.